cilium: panic: interface conversion: *nat.NatKey4 is not tuple.TupleKey: missing method ToHost

On latest master /cc @aanm :
```
# ./cilium/cilium bpf nat list
panic: interface conversion: *nat.NatKey4 is not tuple.TupleKey: missing method ToHost

goroutine 1 [running]:
github.com/cilium/cilium/pkg/maps/nat.(*Map).DumpEntries.func1(0x1bca5e0, 0xc000525170, 0x1bb8420, 0xc0000b46f0)
	/root/go/src/github.com/cilium/cilium/pkg/maps/nat/nat.go:121 +0x57
github.com/cilium/cilium/pkg/bpf.(*Map).DumpWithCallback(0xc00020ee00, 0xc000567b40, 0x0, 0x0)
	/root/go/src/github.com/cilium/cilium/pkg/bpf/map_linux.go:586 +0x500
github.com/cilium/cilium/pkg/maps/nat.(*Map).DumpEntries(0xc00020ee00, 0x1a61b10, 0xc00020ee00, 0x0, 0x0)
	/root/go/src/github.com/cilium/cilium/pkg/maps/nat/nat.go:128 +0x8a
github.com/cilium/cilium/cilium/cmd.dumpNat()
	/root/go/src/github.com/cilium/cilium/cilium/cmd/bpf_nat_list.go:65 +0x221
github.com/cilium/cilium/cilium/cmd.glob..func12(0x2d49620, 0x3307fa0, 0x0, 0x0)
	/root/go/src/github.com/cilium/cilium/cilium/cmd/bpf_nat_list.go:35 +0x3b
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).execute(0x2d49620, 0x3307fa0, 0x0, 0x0, 0x2d49620, 0x3307fa0)
	/root/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2d53540, 0x1, 0x1, 0x2d532e0)
	/root/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:850 +0x2f8
github.com/cilium/cilium/vendor/github.com/spf13/cobra.(*Command).Execute(0x2d53540, 0x407850, 0xc00009c058)
	/root/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/cilium/cilium/cilium/cmd.Execute()
	/root/go/src/github.com/cilium/cilium/cilium/cmd/root.go:47 +0x2d
main.main()
	/root/go/src/github.com/cilium/cilium/cilium/main.go:22 +0x20
```

--- a/pkg/maps/nat/ipv4.go
+++ b/pkg/maps/nat/ipv4.go
@@ -48,7 +48,7 @@ func (n *NatEntry4) String() string {
 }
 
 // Dump dumps NAT entry to string.
-func (n *NatEntry4) Dump(key tuple.TupleKey, start uint64) string {
+func (n *NatEntry4) Dump(key NatKey, start uint64) string {
 	var which string
 
 	if key.GetFlags()&tuple.TUPLE_F_IN != 0 {

--- a/pkg/maps/nat/ipv6.go
+++ b/pkg/maps/nat/ipv6.go
@@ -48,7 +48,7 @@ func (n *NatEntry6) String() string {
 }
 
 // Dump dumps NAT entry to string.
-func (n *NatEntry6) Dump(key tuple.TupleKey, start uint64) string {
+func (n *NatEntry6) Dump(key NatKey, start uint64) string {
 	var which string
 
 	if key.GetFlags()&tuple.TUPLE_F_IN != 0 {

--- a/pkg/maps/nat/nat.go
+++ b/pkg/maps/nat/nat.go
@@ -67,7 +67,7 @@ type NatEntry interface {
 	ToHost() NatEntry
 
 	// Dumps the Nat entry as string.
-	Dump(key tuple.TupleKey, start uint64) string
+	Dump(key NatKey, start uint64) string
 }
 
 // NatDumpCreated returns time in seconds when NAT entry was created.
@@ -118,7 +118,7 @@ func (m *Map) DumpEntries() (string, error) {
 
 	nsecStart, _ := bpf.GetMtime()
 	cb := func(k bpf.MapKey, v bpf.MapValue) {
-		key := k.(tuple.TupleKey)
+		key := k.(NatKey)
 		if !key.ToHost().Dump(&buffer, false) {
 			return
 		}
@@ -148,10 +148,9 @@ func statStartGc(m *Map) gcStats {
 func doFlush4(m *Map) gcStats {
 	stats := statStartGc(m)
 	filterCallback := func(key bpf.MapKey, _ bpf.MapValue) {
-		currentKey := key.(*tuple.TupleKey4Global)
-		err := m.Delete(currentKey)
+		err := m.Delete(key)
 		if err != nil {
-			log.WithError(err).WithField(logfields.Key, currentKey.String()).Error("Unable to delete CT entry")
+			log.WithError(err).WithField(logfields.Key, key.String()).Error("Unable to delete CT entry")
 		} else {
 			stats.deleted++
 		}
@@ -163,10 +162,9 @@ func doFlush4(m *Map) gcStats {
 func doFlush6(m *Map) gcStats {
 	stats := statStartGc(m)
 	filterCallback := func(key bpf.MapKey, _ bpf.MapValue) {
-		currentKey := key.(*tuple.TupleKey6Global)
-		err := m.Delete(currentKey)
+		err := m.Delete(key)
 		if err != nil {
-			log.WithError(err).WithField(logfields.Key, currentKey.String()).Error("Unable to delete CT entry")
+			log.WithError(err).WithField(logfields.Key, key.String()).Error("Unable to delete CT entry")
 		} else {
 			stats.deleted++
 		}

